lib/src/modules/conditional.dart

```diff
@@ -890,6 +889,71 @@ enum ConditionalType {
   priority
 }
 
+/// Shorthand for a [Case] inside a [Conditional] block.
+///
+/// It is used to assign a signal based on a condition with multiple cases to
+/// consider. For e.g., this can be used instead of a nested [mux].
+///
+/// The result is of type [Logic] and it is determined by conditionaly matching
+/// the expression with the values of each item in conditions. If width of the
+/// input is not provided, then the width of  the result is inferred from the
+/// width of the entries.
+Logic cases(Logic expression, Map<dynamic, dynamic> conditions,
+    {int? width, ConditionalType conditionalType = ConditionalType.none}) {
+  for (final condition in conditions.entries) {
+    int? inferredWidth;
+
+    if (condition.value is Logic) {
+      inferredWidth = (condition.value as Logic).width;
+    } else if (condition.value is LogicValue) {
+      inferredWidth = (condition.value as LogicValue).width;
+    }
+
+    if (width != inferredWidth && width != null && inferredWidth != null) {
+      throw SignalWidthMismatchException.forDynamic(
+          condition.value, width, inferredWidth);
+    }
+
+    width ??= inferredWidth;
+
+    if (width != inferredWidth && inferredWidth != null) {
+      throw SignalWidthMismatchException.forDynamic(
+          condition.value, width!, inferredWidth);
+    }
+    if (condition.key is Logic) {
+      if (expression.width != (condition.key as Logic).width) {
+        throw SignalWidthMismatchException.forDynamic(
+            condition.key, expression.width, (condition.key as Logic).width);
+      }
+    }
+
+    if (condition.key is LogicValue) {
+      if (expression.width != (condition.key as LogicValue).width) {
+        throw SignalWidthMismatchException.forDynamic(condition.key,
+            expression.width, (condition.key as LogicValue).width);
+      }
+    }
+  }
+
+  final result = Logic(name: 'result', width: width!);
```

Comment:
why can't `width` be null here?  what if none of the inputs provided something to infer the width?  we probably need a check before this to ensure that `width` is provided if it couldnt be inferred.  also means we're missing a test